### PR TITLE
[BUG] skip intersect iterator qsort if INORDER flag is used

### LINKED ISSUE #1986: Wrong result on slop query
With a slop set to 0, this query shouldn't return any document

```
127.0.0.1:6379> FT.CREATE idx ON HASH SCHEMA label TEXT
OK
127.0.0.1:6379> HSET doc:1 label "Hello the world"
(integer) 1
127.0.0.1:6379> FT.SEARCH idx '(Hello world) => {$slop: 0}'
1) (integer) 1
``` 

```
======================================================================
FAIL: testClient (__main__.RedisSearchTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/circleci/repo/test/test.py", line 169, in testClient
    self.assertEqual(3,client.search(Query('henry king').slop(0).in_order()).total)
AssertionError: 3 != 52
```
https://app.circleci.com/pipelines/github/RediSearch/redisearch-py/654/workflows/7dcd6c25-6c3e-44ec-8eb5-cef51f72eda0/jobs/1325

### PR DESCRIPTION
**Pull request by ashtul**
fixes #1986

### CODE REVIEW

File: src/index.c [hunk 1]
```diff
@@ -781,7 +781,9 @@ IndexIterator *NewIntersecIterator(IndexIterator **its_, size_t num, DocTable *d
   ctx->num = num;
 
   // Sort children iterators from low count to high count which reduces the number of iterations.
-  qsort(ctx->its, ctx->num, sizeof(*ctx->its), (CompareFunc)cmpIter);
+  if (!ctx->inOrder) {
```
Comment (MeirShpilraien) [inline src/index.c:784]: @ashtul would it be better to save 2 arrays, one is sorted to be optimized and the other is for check slop and inorder?

Comment (ashtul) [reply src/index.c:784]: @MeirShpilraien I considered it, but we check for `INORDER` in the same order in which we picked the results from the sub-iterators (`RSIndexResult_HasOffsets(r->children[i])`).
Therefore, the change suggested requires sorting results according to the original order for each matching intersect. So while it is doable, it seems like a complexity that (1) might cost more than whatever was optimized and (2) would be hard and unintuitive to maintains. Removing the optimization when INORDER is used seems like an adequate and elegant solution.

WDYT?

File: tests/pytests/test.py [hunk 2]
```diff
@@ -975,6 +975,20 @@ def testSlopInOrder(env):
     env.assertEqual(0, r.execute_command(
         'ft.search', 'idx', 't1 t2 t3 t4', 'inorder')[0])
 
+    # test with qsort optimization on intersect iterator
```
Comment (MeirShpilraien) [inline tests/pytests/test.py:978]: I also suggest creating another test for it (maybe with the issue number).
When tests get large it's harder to spot what went wrong when they failed.

**Review by MeirShpilraien — APPROVED** [review]
I believe it's a good fix, but I think we should open another issue to enable this optimization even if in order is used...

**Pull request merged**